### DISCUSSION TITLE
Enable autoApprove for bitbucket-server

### DISCUSSION THREAD

**Discussion by Manligheten** [renovatebot/renovate#25370] [Suggest an Idea]
**What would you like Renovate to be able to do?**
I would like Renovate to be able to automatically approve a PR it has created in Bitbucket, similar to how it is done for Azure or Gitlab platforms.

**If you have any ideas on how this should be implemented, please tell us here.**
Looking at how PRs is created or updated in azure, similar could be done for the Bitbucket-server platform.
That is, by checking the `autoApprove` platform option.

**Is this a feature you are interested in implementing yourself?**

Maybe

**Comment by rarkins** [comment]
Using which API and which permissions?

**Reply by Manligheten** [reply]
We are running Bitbucket Server v8.x, so I suspect its this one [create pull request](https://developer.atlassian.com/server/bitbucket/rest/v815/api-group-pull-requests/#api-api-latest-projects-projectkey-repos-repositoryslug-pull-requests-post).

I am not sure about required permissions, how would I verify those? I am thinking as our bot is allowed to create PRs, it should be able to also include itself as a approving participant(?)

**Reply by rarkins** [reply]
This requires much more research. It sounds like you're not sure if it can even be done, so it's premature to be asking for it as a feature

**Reply by Manligheten** [reply]
I was indeed a bit fooled by the API, as I just guessed that adding participants with the "approved" field set to true would suffice.
After some basic testing it seems like Bitbucket-server does not allow for:

1. Assigning yourself as a reviewer to your own PR.
2. Approving your own PR.

So this request is indeed dead in the water, I will have to let another bot-user automatically approve some PRs created by Renovate.

Thank you!

**Reply by viceice** [reply]
the same restrictions are enforced on gitea and GitHub
